webui/react/src/components/kit/Tags.tsx

```diff
@@ -174,4 +179,21 @@ const EditableTagList: React.FC<Props> = ({
   );
 };
 
-export default EditableTagList;
+export default Tags;
+
+export const tagsActionHelper = (
```

Comment:
i don't think i understand why we need this exported function.

if the `Tag` component can receive the tags and a callback fn as props, can't we just do the push/filter/replace operations within that component, and still call the callback fn with the updated tags there?

I might be missing something, but if we're adding the complexity of having to import `tagsActionHelper` everywhere we use the component, and redundantly passing `tags` to the component twice, I just want to make sure it's worth that trade-off.

Reply:
The idea here is that eventually we want to change the tags API to take parameters like `{addTag: string, removeTag: string, updateTag: {id, string}}`, because passing the entire list of tags to the backend can create race condition.

Reply:
we discussed this on a call. recap:

`tagsActionHelper` was intended to allow the component to only call `onAction`, let the `onChange` prop be removed, and still pass the updated list of tags in the helper: `callbackFn(newTags)` 

eventually we want to stop using API calls that take the updated list of tags, and replace them with API calls that take only the updated tag. At that point, the `tagsActionHelper` could be removed.

perhaps a comment that briefly explains this would be helpful?

also, personally i still think i'd prefer to have the `Tags` component just continue using both `onAction` and `onChange`, without `tagsActionHelper`, and eventually transition to deprecating the `onChange` prop. IMO feels cleaner than having to use the helper and removing it later. but avoiding future changes to the component itself makes sense, too, so I don't feel that strongly about it.